### PR TITLE
f-strings, again

### LINKED ISSUE #2870: F-string still aren't working correctly.
The #2829 was a fix for #2822. But it handles only a simple case where the variable is the only thing inside braces which is a small subset of possible usage. The code inside braces might be almost any valid xonsh expression.
The #2829 also breaks some things. For example, the first two cases below which is escaping braces with double braces and using f-strings inside an f-string.

Here I provide some cases that aren't currently handled correctly:
```
f"{{$X}}"
f"{f'{$X}'}"
f"{$X + $Y}"
f"{${'X'}}"
f"{repr($X)}"
f"{$(uname)}"
```

Those cases _are_ currently handled correctly but might be useful for testing:
```
f"{'$X'}"
f"a{$X}b"
f"{'}'}"
f"{'$'}"
```

As mentioned in #2829 we might need to parse the f-string content to correctly handle all possible cases. As a reference for implementation, we could check the f-string related code from the cpython's [parser implementation](https://github.com/python/cpython/blob/656d52dbfde3223cd2a3525d652b6cccb02fa991/Python/ast.c#L4927).

### PR DESCRIPTION
**Pull request by scopatz**
This fixes #2870, and also updates the news format to the newer format which confuses fewer people with the None removal

### CODE REVIEW

File: news/fstrenv.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* f-strings (``f"{expr}"``) are now fully capable of executing xonsh expressions.
+  The one exception to this is that ``![cmd]`` and ``!(cmd)`` do work because
```
Comment (MrQubo) [inline news/fstrenv.rst]:
```suggestion
  The one exception to this is that ``![cmd]`` and ``!(cmd)`` don't work because
```
Should be `do NOT work`.

Comment (scopatz) [reply news/fstrenv.rst]: nice catch!

**Comment by scopatz** [comment]
I think this is ready now!

**Comment by scopatz** [comment]
Thanks @gforsyth! I added some CLI options to force pip to upgrade, which seems to have worked.

**Comment by gforsyth** [comment]
Looks good to me! Thanks, @scopatz!

**Pull request merged**